### DISCUSSION TITLE
Microsoft vsdbg license

### DISCUSSION THREAD

**Discussion by ghost** [coder/code-server#2206] [General] [answered]
I'm not expecting this to be possible, but it's not very useful to be able to code C# in my browser if I can't also debug it.

```
.NET Debugging is only supported in Microsoft versions of VS Code
```

Obviously Microsoft still holds the keys here. What can we do? I still have a remote desktop environment that I do most of the work on.

**Comment by code-asher** [comment, marked as answer]
I think the only workaround is to use netcoredbg instead of vsdbg. Check
out https://github.com/cdr/code-server/issues/1942#issuecomment-669313941 for more information.

One bug with this is that using Console.ReadLine() doesn't seem to work:
https://github.com/cdr/code-server/issues/1957
